OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -204,26 +204,29 @@ protected void ScheduleDelayedAction(int t, Action a)
 				a();
 		}
 
-		// Note: facing is only used by the legacy positioning code
-		// The world coordinate model uses Actor.Orientation
-		public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
+		protected virtual bool BarrelCanFire(Actor self, Target target)
```

Comment:
I would expect a method named `BarrelCanFire` to have something to do with a barrel?